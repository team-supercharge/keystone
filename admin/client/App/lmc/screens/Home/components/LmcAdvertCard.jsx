import React, { Component } from 'react';
import PropTypes from 'prop-types';

class LmcAdvertCard extends Component {
    render () {
        const { url, image } = this.props;
        const defaultImage = 'https://s3.eu-west-2.amazonaws.com/lmc-data-production/public/careoffice-home-screen-advert.png';
        const defaultUrl = 'https://calendly.com/logmycare';
        return (
            <div >
                <h2 className="lmc-card-title" >
                    { TITLE }
                </h2>
                <div className="lmc-card" style={styles.container}>
                    <a href={url || defaultUrl} target="_blank">
                        <img style={{ width: '102%' }} src={image || defaultImage} alt=""/>
                    </a>
                </div>
            </div>
        );
    }
}

const styles = {
    container: {
        overflow: 'hidden',
        justifyContent: 'center'
    }
}
const TITLE = 'What\'s on offer';
LmcAdvertCard.propTypes = {
    image: PropTypes.string,
    url: PropTypes.string,
};

export default LmcAdvertCard;
