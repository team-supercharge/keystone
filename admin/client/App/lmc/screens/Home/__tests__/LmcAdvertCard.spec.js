import React from 'react'
import { shallow } from 'enzyme'
import LmcAdvertCard from '../components/LmcAdvertCard'

describe('LmcAdvertCard', () => {
    let wrapper
    let url
    let img

    beforeEach(() => {
        url = 'TestUrl'
        img = 'TestImage'

        wrapper = shallow((
            <LmcAdvertCard
                url={url}
                image={img}
            />
        ));
    })
    test('should render correct snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    })

    test('should have correct link', () => {
        const link = wrapper.find('a')
        expect(link.props().href).toEqual(url)
    })

    test('should have correct image', () => {
        const image = wrapper.find('img')
        expect(image.props().src).toEqual(img)
    })
})