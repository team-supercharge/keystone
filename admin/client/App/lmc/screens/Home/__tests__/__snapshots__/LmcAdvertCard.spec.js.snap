// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcAdvertCard should render correct snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <h2
      className="lmc-card-title"
    >
      What's on offer
    </h2>
    <div
      className="lmc-card"
      style={
        Object {
          "justifyContent": "center",
          "overflow": "hidden",
        }
      }
    >
      <a
        href="TestUrl"
        target="_blank"
      >
        <img
          alt=""
          src="TestImage"
          style={
            Object {
              "width": "102%",
            }
          }
        />
      </a>
    </div>
  </div>,
  "nodes": Array [
    <div>
      <h2
        className="lmc-card-title"
      >
        What's on offer
      </h2>
      <div
        className="lmc-card"
        style={
          Object {
            "justifyContent": "center",
            "overflow": "hidden",
          }
        }
      >
        <a
          href="TestUrl"
          target="_blank"
        >
          <img
            alt=""
            src="TestImage"
            style={
              Object {
                "width": "102%",
              }
            }
          />
        </a>
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcAdvertCard
        image="TestImage"
        url="TestUrl"
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcAdvertCard {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "image": "TestImage",
          "url": "TestUrl",
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <h2
            className="lmc-card-title"
          >
            What's on offer
          </h2>
          <div
            className="lmc-card"
            style={
              Object {
                "justifyContent": "center",
                "overflow": "hidden",
              }
            }
          >
            <a
              href="TestUrl"
              target="_blank"
            >
              <img
                alt=""
                src="TestImage"
                style={
                  Object {
                    "width": "102%",
                  }
                }
              />
            </a>
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <h2
            className="lmc-card-title"
          >
            What's on offer
          </h2>
          <div
            className="lmc-card"
            style={
              Object {
                "justifyContent": "center",
                "overflow": "hidden",
              }
            }
          >
            <a
              href="TestUrl"
              target="_blank"
            >
              <img
                alt=""
                src="TestImage"
                style={
                  Object {
                    "width": "102%",
                  }
                }
              />
            </a>
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcAdvertCard
    image="TestImage"
    url="TestUrl"
  />,
}
`;
